skll/experiments/__init__.py

```diff
@@ -356,55 +415,79 @@ def _classify_featureset(args):  # noqa: C901
+                    grid_scores = [results[0]]
+                    grid_search_cv_results_dicts = [results[1]]
+                    if grid_search:
+                        logger.info(f"Best {grid_objective} grid search score: "
+                                    f"{round(results[0], 3)}")
+
+                # save model, if asked
                 if model_path:
                     learner.save(modelfile)
 
-                if grid_search:
-                    logger.info(f"Best {grid_objective} grid search score: "
-                                f"{round(best_score, 3)}")
-            else:
-                grid_scores = [None]
-                grid_search_cv_results_dicts = [None]
-
-            # print out the parameters
+            # print out the model parameters; note that for
+            # voting learners, we exclude the parameters for
+            # the underlying estimators
+            params = learner.model.get_params(deep=isinstance(learner, Learner))
             param_out = (f'{param_name}: {param_value}' for
-                         param_name, param_value in
-                         learner.model.get_params().items())
+                         param_name, param_value in params.items())
             logger.info(f"Hyperparameters: {', '.join(param_out)}")
 
-            # run on test set or cross-validate on training data,
-            # depending on what was asked for
+            # evaluate the model on the test set;
             if task == 'evaluate':
                 logger.info("Evaluating predictions")
+                # voting learners have an extra keyword argument indicating
+                # whether we want to save the individual predictions (or votes)
+                extra_kwargs = {}
+                if isinstance(learner, VotingLearner):
+                    extra_kwargs["individual_predictions"] = save_votes
                 task_results = [learner.evaluate(test_examples,
                                                  prediction_prefix=prediction_prefix,
                                                  grid_objective=grid_objective,
-                                                 output_metrics=output_metrics)]
+                                                 output_metrics=output_metrics,
+                                                 **extra_kwargs)]
             elif task == 'predict':
                 logger.info("Writing predictions")
                 # we set `class_labels` to `False` so that if the learner is
-                # probabilistic, probabilities are written instead of labels
+                # probabilistic, probabilities are written instead of labels;
+                # voting learners have an extra keyword argument indicating
+                # whether we want to save the individual predictions (or votes)
+                extra_kwargs = {}
+                if isinstance(learner, VotingLearner):
+                    extra_kwargs["individual_predictions"] = save_votes
```

Comment:
RE: My earlier comment: Could this be done for xval too? Would it be as simple as adding an extra keyword argument?

Reply:
Yes, it can! I just pushed new commits adding this functionality!